Update Jackett to v0.8.490 and to use generic DSM 5+6 services

m4tt075 opened the pull request:
### Motivation
Contains @kaso17 's Jackett updates and fixes of #3032 and adapts the package to use generic DSM 5+6 services

### Linked issues
#3015, #2986, #2881, #1995, Jackett/Jackett#1985, Jackett/Jackett#2217

### Notes
- ~~Depends on merging of @kaso 's Jackett update PR #3032.~~ Squashed into this PR on request of @ymartin59 and with agreement of @kaso17  
- ~~Depends on merging of #3064. Do NOT merge without it or old configurations will be overwritten on upgrades.~~ #3064 merged in the meantime.
- ~~Further testing needed.~~ Outdated.

### Checklist
- [x] Build rule `all-supported` completed successfully on `XPE 5.2` and `XPE 6.1` systems.
- [x] Package upgrade completed successfully
- [x] New installation of package completed successfully

m4tt075 commented:
@kaso17 As to your hints / notes in #3032:
- I have tested the package on an `XPE 5.2` and `XPE 6.1` system. It installs and runs as far as I can judge. At least I can access the web interface in both cases.
- User premissions for the `share` folder seem correct, because I could succesfully upgrade the package using the internal upgrade process on both systems.
- I could not test a migration from a legacy installation as I don't have one, but would rely on the generic service scripts to reset user permissions correctly as this applies to all packages.
- I could verify that all environment variables are set and used correctly.
- The only piece I could not implement from your approach was the logging functionality as this is driven through the generic service scripts. As such, standard logging to `var/jackett.log` is used for now. It works to the extend that both `DSM` and `jackett` log into that file correctly. It can be accessed through the DSM web-UI. However, your (very nice) `log rotation` functionality is lost unfortunately. I don't know how much information is usually written to the logs, but if you foresee an issue here, please let me know.

kianrafiee commented:
newly installed jackett package from @Safihre repo will not start. it was starting fine yesterday and updated. but now it wont start. not sure if there was a jackett update yesterday that broke it

m4tt075 commented:
Not related to this PR or spksrc development. Therefore, please report this in the `Issues` section instead, either in one of the old threads or open a new one.

ymartin59 commented on spk/jackett/Makefile:
```diff
@@ -1,33 +1,38 @@
-MAINTAINER = SynoCommunity
-DESCRIPTION = "Jackett works as a proxy server: it translates queries from apps into tracker-site-specific http queries, parses the html response, then sends results back to the requesting software. Jackett is a single repository of maintained indexer scraping & translation logic - removing the burden from other apps."
-ADMIN_PORT = 9117
+MAINTAINER = 'kaso17 (Jackett), m4tt075 (spksrc integration)'
```
I wonder if `spkrepo` will accept packages with such maintainer. As far as I understood/remember, it has to match a user account registered in registry

ymartin59 commented on spk/jackett/Makefile:
```diff
@@ -1,33 +1,38 @@
 
-INSTALL_PREFIX = /usr/local/$(SPK_NAME)
+# Admin link
+ADMIN_PORT = $(SERVICE_PORT)
```
Is it relevant to consider SERVICE_PORT as ADMIN_PORT ? Probably not, but I prefer to keep it for "backward compatibility"

m4tt075 replied:
Sorry, not sure I understand. Do you want me to remove the comment and the ADMIN_PORT definition or do you want me to keep it?

ymartin59 replied:
It depends if link generated in package view is really usable (often `http://ip:ADMIN_PORT/`) whereas `SERVICE_PORT` is used to provide application link in `app/config` which allows finer settings like `https` and context path in url.
So if "admin link" in package view is not usable as-is, please remove it. If it works, let it there.

m4tt075 replied:
It is usable and works correctly, so I kept it.

ymartin59 commented on spk/jackett/src/service-setup.sh:
```diff
@@ -0,0 +1,65 @@
+    COMMAND="env HOME=${HOME_DIR} PATH=${PATH} LD_LIBRARY_PATH=${SYNOPKG_PKGDEST}/lib ${MONO} --debug ${JACKETT} --PIDFile ${PID_FILE}"
+
+    if [ $SYNOPKG_DSM_VERSION_MAJOR -lt 6 ]; then
+        su ${EFF_USER} -s /bin/sh -c "cd ${SYNOPKG_PKGDEST}; ${COMMAND}" >> ${LOG_FILE} 2>&1 &
```
Is it really needed to change process working directory to SYNOPKG_PKGDEST ? (I did it on `demoservice` to browse files...)
I think HOME_DIR may be more relevant, and it is already done by `su`

ymartin59 commented on spk/jackett/src/service-setup.sh:
```diff
@@ -0,0 +1,65 @@
+    HOME_DIR="${SYNOPKG_PKGDEST}/var"
+
+    echo "Starting Jackett as user ${EFF_USER}" >> ${LOG_FILE}
+    COMMAND="env HOME=${HOME_DIR} PATH=${PATH} LD_LIBRARY_PATH=${SYNOPKG_PKGDEST}/lib ${MONO} --debug ${JACKETT} --PIDFile ${PID_FILE}"
```
Is `--debug` flag really relevant ?

ymartin59 commented on spk/jackett/src/service-setup.sh:
```diff
@@ -0,0 +1,65 @@
+    fi
+}
+
+service_preinst ()
```
Please discard all other hook functions with only logging... I only wrote them for demo, I consider them not relevant for a package

kaso17 commented:
@m4tt075 thank you for taking care of this.
Did you test if the jackett internal updating mechanism works?

In case you didn't:
Please go to the jackett admin panel (port 9117) and click on the yellow "Check for updates" button. Wait a little until jackett updates itself and refresh the page. At the bottom of the page it should say version 0.8.539.0 (or newer).

I don't see that the bin/share directory `${SYNOPKG_PKGDEST}/share/${SYNOPKG_PKGNAME}` is chmod/chowned anywhere anymore. If the jackett user can't write to that directory the update won't work. I assume the /var directory is chowned to the new user during the update? Besides that I don't see any obvious issues.

m4tt075 commented:
@kaso17 All fine. Thanks for reviewing...

ymartin59 commented on spk/jackett/src/service-setup.sh:
```diff
@@ -0,0 +1,35 @@
+    if [ $SYNOPKG_DSM_VERSION_MAJOR -lt 6 ]; then
+        su ${EFF_USER} -s /bin/sh -c "${COMMAND}" >> ${LOG_FILE} 2>&1 &
+    else
+        cd ${SYNOPKG_PKGDEST}
```
You removed "cd" in `COMMAND` for DSM 5 and so it should be removed here for DSM 6 too.

m4tt075 replied:
Removed.

m4tt075 commented:
@ymartin59 I have addressed your latest comments. Anything else?

ward0 commented:
When can we see an update pls? @m4tt075

m4tt075 commented:
@ymartin59 @kaso17 Following up with your recent comments in #3032, I have...
- rebased
- squashed all commits (including @kaso17 's) into one
- Adapted the PR title
- Adapted the PR description to keep the linked issues from @kaso17 's PR description
- Re-compiled Jackett for bromolow architecture 5.2 and 6.1, just to make sure
- Successfully upgraded the package on XPE 6.1 within the package manager and then again, using Jackett's internal upgrading process
- Successfully installed the package on XPE 5.2 and upgraded it using Jackett's internal upgrading process

Should be ready to go now...

ymartin59 commented:
Marvelous. Merging.
By the way, online repository rejects mono 5.4 package for x64... so it may not be published in a short term.

ymartin59 commented on spk/jackett/src/service-setup.sh:
```diff
@@ -0,0 +1,34 @@
+{
+    # Replace generic service startup
+
+    MONO_PATH="${SYNOPKG_PKGDEST}/../mono/bin"
```
Bad idea... Prefer to use `/usr/local/mono/bin`

ymartin59 commented on spk/jackett/src/service-setup.sh:
```diff
@@ -0,0 +1,34 @@
+    echo "Starting Jackett as user ${EFF_USER}" >> ${LOG_FILE}
+    COMMAND="env HOME=${HOME_DIR} PATH=${PATH} LD_LIBRARY_PATH=${SYNOPKG_PKGDEST}/lib ${MONO} ${JACKETT} --PIDFile ${PID_FILE}"
+
+    if [ $SYNOPKG_DSM_VERSION_MAJOR -lt 6 ]; then
```
These if statements are same as generic script.
I think moving all variables settings out of `service_prestart` (so removing this function) and provide `SERVICE_COMMAND` at the begging of this script should work.

Safihre replied:
No that won't work because the process never detaches: no `&` at the end when using SERVICE_COMMAND` for processes that don't support deamon.
So DSM will stay in a constant state of `Starting` since the `start` never returns.
So that's why we need this special function. Or integrated in generic approach :)

m4tt075 replied:
*LOL* Just came here to write the same! Agree with @Safihre . This is another candidate for the `start-stop-daemon` approach. I propose to keep this package as is for the time being and update once that approach is established...

m4tt075 commented:
@ymartin59 This has been merged 6 weeks ago. Anything preventing this from being published?

ymartin59 commented:
@m4tt075 Yes... Mono update. The fix I designed to relocate `cert-sync` seems to work. My DSM has a pending update, I will test everything is ok with mono+jackett upgrade

Safihre commented:
@jdeluyck Could you restart your NAS and try again?
What model NAS do you use?

jdeluyck commented:
I'm using a DS916+. 
Rebooted NAS, retried upgrade, same story.

ymartin59 commented:
Hum hum. Corrupt package ? It remembers me trouble with signing process in repository... I have already fixed it (and there was no issue with recent published packages like mono, ffmpeg...) so probably there is a specific failure with Jackett. I am investigating....

ymartin59 commented:
@jdeluyck As an immediate work-around, open Package Center "Settings" and select "Any publishers" in Trust Level section... sorry for the inconvenience

Safihre commented:
Simple re-publish is not fixing it? I indeed also see this error..

jdeluyck commented:
@ymartin59 that indeed works. So the package is ok, signing is borked.
